src/runtime_src/core/pcie/tools/xbutil/xbutil.h

```diff
@@ -148,6 +148,53 @@ static const std::map<MEM_TYPE, std::string> memtype_map = {
 
 static const std::map<std::string, command> commandTable(map_pairs, map_pairs + sizeof(map_pairs) / sizeof(map_pairs[0]));
 
+class xclbin {
+public:
+        xclbin(unsigned index, unsigned cu_idx)
+        try 
+        {
+
+            m_idx = index;
+            m_cuidx = cu_idx;
+            m_handle = nullptr;
+
+            std::string devstr = "device[" + std::to_string(m_idx) + "]";
+            m_handle = xclOpen(m_idx, NULL, XCL_QUIET);
+            if (!m_handle)
+                throw std::runtime_error("Failed to open " + devstr);
+
+            std::string xclbinid, errmsg;
+            uuid_t uuid;
+            pcidev::get_dev(m_idx)->sysfs_get("", "xclbinuuid", errmsg, xclbinid);
+
+            if (!errmsg.empty())
+                throw std::runtime_error(errmsg);
+
+            uuid_parse(xclbinid.c_str(), uuid);
+
+            if (uuid_is_null(uuid))
+                throw std::runtime_error("'uuid' invalid, please re-program xclbin.");
+
+            uuid_copy(m_uuid, uuid);
+
+            if (xclOpenContext(m_handle, m_uuid, m_cuidx, true))
+                throw std::runtime_error("ERROR: Unable to lockdown xclbin.");
+       
+        } catch (std::runtime_error) {
```

Comment:
Never catch by value.  Do "catch (const std::exception& ex) {" 
You should rethrow the exception, otherwise you end up with a bogus object.
There is nothing to close if handle or context wasn't successfully opened.   While you need to close the handle, it doesn't appear there can ever be a context to close.

With proper device object passed to constructor there is no need for try / catch in the xclbin ctor.